### ISSUE TITLE
Allow specifying encode mode so all images are the same

### ISSUE THREAD

**Issue by hvanbakel** [codebude/QRCoder#81]
I'm generating an animated GIF that has to have all equally sized images, the codes that I use to generate images for are randomly generated.

I ran into issues though if one of the codes contained only numerics and/or upper case letters because [QRCodeGenerator](https://github.com/codebude/QRCoder/blob/6defbee8c6d8ca27e4afeea04b52467ab8f537cf/QRCoder/QRCodeGenerator.cs#L745-746) assumes that it is alphanumeric thus generating a different image size.

In order for the animated gif to work properly, it needs to have same sizes for all of them.

**Comment by Hydroque** [comment]
Could you briefly specify the problem in one statement on what would resolve this issue for sake of ease of read?

**Comment by codebude** [comment]
@Hydroque I think the problem of @hvanbakel is the following:
QrCoder or better said the QrCode class has only "pixelsPerModule" in the GetGraphic-method to influence the size of the QrCode image. If he encodes only numeric data QrCoder generates a code with less modules (black and white boxes) and the QrCode image gets a smaller resolution. 

So what he misses is an overload of the GetGraphics method, which has parameters not for "pixelsPerModule", but for "fixedOutputWidth" and "fixedOutputHeight". Maybe I'll add this in one of the next releases, but right now I have no time.

For a quick solution @hvanbakel could meausre the side length of every QrCode generated and if it's too small, he could resize the QrCode graphic to the desired size.

**Comment by hvanbakel** [comment]
Exactly, my workaround is to just use a lowercase letter as the last char
in the sequence that I generate to be put into the qr code. Besides this
issue it works great, nice work!

**Comment by nitrovent** [comment]
@hvanbakel do you need all renderings having the same module count or is it sufficient that the resulting image width stays the same?

**Comment by hvanbakel** [comment]
I believe the only issue is that the image width varied, as a result of that the animated gif could not be generated. I might be able to write a unit test that illustrates the problem. As far as the solution goes, I’m not familiar enough with the code to properly answer your question.

**Comment by codebude** [comment]
@nitrovent Same module count would produce images with the same width (because each module is rendered with a fixed pixel width). @hvanbakel 's problem was that he wanted to create one .gif file out of multiple QR code images. Since one gif file can only have one resolution, most GIF classes expect all frame (e.g. all QR code images) to be in the same dimensions. 
If I understood @hvanbakel correct, some of his QR codes had only numbers in the payload, while others had chars. Since the lib uses the possibly most efficient enconding mode, it encodes numeric only payloads in a more efficient way, which produces less modules and thus graphics/QR images with smaller dimensions.

To avoid @hvanbakel problem (namely different image dimensions for fixed length payloads) I see multiple solutions:
- Add one UTF8 char into the payload, to force the library to choose Byte-Mode enconding (which results in same size QR images for same length input payloads. --> That is what @hvanbakel did
- Change the library to set a fixed enconding mode and not let the library choose the most efficient way. --> Is part of this commit https://github.com/codebude/QRCoder/pull/111/commits/77c496f921359a49e37868155a2e90a79ac2c159 from this pull request: https://github.com/codebude/QRCoder/pull/111 . If I merge it, one could use the EciMode parameter.
- Take the library at it is, take the efficient encoding modes and just use C# Graphics-class to re-scale the outcoming images to unified dimensions, so that the Gif writer acceots them.

**Comment by hvanbakel** [comment]
@codebude that’s exactly what my problem is, thanks for filling in my blanks on the image side 👍

**Comment by nitrovent** [comment]
Thanks for the quick answer. 
Maybe my question was a bit unclear. I should rather have asked if it is (from a visual aesthetic point) necessary to have the same module count in each rendering (that would favor option 2 - fixed encoding mode) or if it would be sufficient to have an option specify a total width and let the QrCoder calculate the corresponding pixelsPerModule (close to option 3 but would yield sharper output i guess).
